unicodecsv/py2.py

```diff
@@ -112,20 +134,28 @@ def __init__(self, f, dialect=None, encoding='utf-8', errors='strict',
         self.encoding_errors = errors
         self._parse_numerics = bool(
             self.dialect.quoting & csv.QUOTE_NONNUMERIC)
+        self._current_row_number = 0
 
     def next(self):
+        self._current_row_number += 1
         row = self.reader.next()
         encoding = self.encoding
         encoding_errors = self.encoding_errors
-        unicode_ = unicode
-        if self._parse_numerics:
-            float_ = float
-            return [(value if isinstance(value, float_) else
-                    unicode_(value, encoding, encoding_errors))
-                    for value in row]
-        else:
-            return [unicode_(value, encoding, encoding_errors)
-                    for value in row]
+        parse_numerics = self._parse_numerics
+        values = []
+        try:
+            for column, value in enumerate(row, start=1):
+                values.append(
+                    value
+                    if parse_numerics and isinstance(value, float) else
+                    unicode(value, encoding, encoding_errors)
+                )
+        except UnicodeDecodeError as error:
+            error.message = 'Value in cell {}{} can not be decoded. {}'.format(
+                _get_column_name(column), self._current_row_number, str(error)
+            )
```

Comment:
This error message seems pretty tied to understanding csv the way it's presented in excel. Perhaps we can say "column X of row Y"?